src/main/java/org/broadinstitute/hellbender/tools/exome/ExomeReadCounts.java

```diff
@@ -0,0 +1,681 @@
+import org.broadinstitute.hellbender.utils.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.*;
+
+/**
+ * Calculate read-counts on an exome given its intervals.
+ *
+ * @author Valentin Ruano-Rubio &lt;valentin@broadinstitute.org&gt;
+ */
+@CommandLineProgramProperties(
+        usage = "Count overlapping reads exon by exon",
+        usageShort = "Count overlapping reads exon by exon",
+        programGroup = ExomeAnalysisProgramGroup.class
+)
+@SuppressWarnings("unused")
+public class ExomeReadCounts extends ReadWalker {
+
+
+    /**
+     * Default cohort name used unless one is provided using argument {@link #cohortName}.
+     */
+    public static final String DEFAULT_COHORT_NAME = "<ALL>";
+
+
+    public static final String ANONYMOUS_SAMPLE_ID = "<anonymous>";
+    public static final String ANONYMOUS_READ_GROUP_ID = "<anonymous>";
+
+
+    /**
+     * Possible read count grouping.
+     */
+    public enum GroupBy {
+        /**
+         * Collapse all sample counts into one.
+         */
+        COHORT,
+
+        /**
+         * Count per sample.
+         */
+        SAMPLE,
+
+        /**
+         * Count per read group.
+         */
+        READ_GROUP;
+    }
+
+    @Argument(doc="output tabular file with the counts",shortName="o",optional=false)
+    protected File output = null;
+
+    @Argument(doc="summary per column output tabular file",shortName="co",optional=true)
+    protected File summaryPerColumnOutput = null;
+
+    @Argument(doc="summary per row output tabular file",shortName="ro",optional=true)
+    protected File summaryPerRowOutput = null;
+
+    @Argument(doc="group counts by Cohort (all samples in one), Sample or ReadGroup", shortName="groupBy", fullName="groupBy", optional=true)
+    protected GroupBy groupBy = GroupBy.COHORT;
+
+    @Argument(doc="Cohort name used to name the read count column when the groupBy argument is set to COHORT (default)",optional=true,fullName="cohortName",shortName="cohort")
+    protected String cohortName = DEFAULT_COHORT_NAME;
+
+    @Argument(doc="Format use to name count column headers. It can be any string where the following special  " +
+            "sequences would be substituted: %^ -> default name given the groupBy option, %s -> sample-id, %r -> read-group-id, %c -> cohort name", optional=true,
+       fullName="countColumnName",shortName="countColumnName")
+    protected String countColumnName = "%^";
+
+    private PrintWriter outputWriter;
+
+    private PrintWriter summaryPerColumnOutputWriter;
+
+    private PrintWriter summaryPerRowOutputWriter;
+
+    private SampleDB sampleDB;
+
+    private int groupCount = 1;
+
+    private static final Logger logger = LogManager.getLogger(ExomeReadCounts.class);
+
+    private ExonDB<GenomeLoc> exonDB;
+
+    private Counts counts;
+
+    private GenomeLocParser genomeLocFactory;
+
+    @Override
+    public void onTraversalStart() {
+        super.onTraversalStart();
```

Comment:
fine. but unused code is just overhead for the reader because now i have to read the super method and then i find out it's empty so I lost time and context.